apps/meteor/app/file-upload/server/methods/sendFileMessage.ts

```diff
@@ -9,6 +9,101 @@ import { FileUpload } from '../lib/FileUpload';
 import { canAccessRoom } from '../../../authorization/server/functions/canAccessRoom';
 import { SystemLogger } from '../../../../server/lib/logger/system';
 
+export const parseFileIntoMessageAttachments = async (
+	file: Record<string, any>,
+	roomId: string,
+	user: IUser,
+): Promise<Record<string, any>> => {
+	await Uploads.updateFileComplete(file._id, user._id, _.omit(file, '_id'));
+
+	const fileUrl = FileUpload.getPath(`${file._id}/${encodeURI(file.name)}`);
+
+	const attachments: MessageAttachment[] = [];
+
+	const files = [
+		{
+			_id: file._id,
+			name: file.name,
+			type: file.type,
+		},
+	];
+
+	if (/^image\/.+/.test(file.type)) {
+		const attachment: FileAttachmentProps = {
+			title: file.name,
+			type: 'file',
+			description: file.description,
+			title_link: fileUrl,
+			title_link_download: true,
+			image_url: fileUrl,
+			image_type: file.type,
+			image_size: file.size,
+		};
+
+		if (file.identify?.size) {
+			attachment.image_dimensions = file.identify.size;
+		}
+
+		try {
+			attachment.image_preview = await FileUpload.resizeImagePreview(file);
+			const thumbResult = await FileUpload.createImageThumbnail(file);
+			if (thumbResult) {
+				const { data: thumbBuffer, width, height } = thumbResult;
+				const thumbnail = FileUpload.uploadImageThumbnail(file, thumbBuffer, roomId, user._id);
+				const thumbUrl = FileUpload.getPath(`${thumbnail._id}/${encodeURI(file.name)}`);
+				attachment.image_url = thumbUrl;
+				attachment.image_type = thumbnail.type;
+				attachment.image_dimensions = {
+					width,
+					height,
+				};
+				files.push({
+					_id: thumbnail._id,
+					name: file.name,
+					type: thumbnail.type,
+				});
+			}
+		} catch (e) {
+			SystemLogger.error(e);
+		}
+		attachments.push(attachment);
+	} else if (/^audio\/.+/.test(file.type)) {
+		const attachment: FileAttachmentProps = {
+			title: file.name,
+			type: 'file',
+			description: file.description,
+			title_link: fileUrl,
+			title_link_download: true,
+			audio_url: fileUrl,
+			audio_type: file.type,
+			audio_size: file.size,
+		};
+		attachments.push(attachment);
+	} else if (/^video\/.+/.test(file.type)) {
+		const attachment: FileAttachmentProps = {
+			title: file.name,
+			type: 'file',
+			description: file.description,
+			title_link: fileUrl,
+			title_link_download: true,
+			video_url: fileUrl,
+			video_type: file.type,
+			video_size: file.size,
+		};
+		attachments.push(attachment);
+	} else {
+		const attachment = {
+			title: file.name,
+			type: 'file',
+			description: file.description,
+			title_link: fileUrl,
+			title_link_download: true,
+		};
+		attachments.push(attachment);
+	}
+	return { files, attachments };
+};
+
 Meteor.methods({
 	async sendFileMessage(roomId, _store, file, msgData = {}) {
```

Comment:
The value and format of the `file` param is not validated anywhere on this method, please include a validation to ensure proper data is sent.